### PR TITLE
Feature/daef 241 react polymorph component for floating point numbers

### PR DESCRIPTION
**Pull request by tomothespian**
This PR introduces NumericInput component.

screen shot 2017-05-26 at 18 48 05

screen shot 2017-05-26 at 18 48 16

### CODE REVIEW

**Comment by DominikGuzei** [comment]
i am reviewing this now

**Comment by DominikGuzei** [comment]
Great work, @tomothespian - this already is much more solid than what we had before!

Here is one UX issue that feels a bit awkward currently:
- Input `99.000000` into the field and then move the cursor after the first zero. Then hit the "dot" key. **Expected**: the dot should move behind the first zero. **Actual**: nothing happens.

File: source/components/NumericInput.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+  static propTypes = Object.assign({}, FormField.propTypes, {
+    error: PropTypes.string,
+    value: PropTypes.string,
+    placeholder: PropTypes.string,
+    maxLength: PropTypes.number,
+    maxBeforeDot: PropTypes.number,
+    maxAfterDot: PropTypes.number,
+    maxValue: PropTypes.number,
+    rightLabel: PropTypes.string,
```
Comment (DominikGuzei) [inline source/components/NumericInput.js]: I think we should remove the `rightLabel` feature, as it is very application specific to Daedalus and does not necessarily belong to a general-purpose numeric input. In Daedalus we should create a special component and add the right label feature there.

File: source/components/NumericInput.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import Input from './Input';
+import FormField from './FormField';
+
+export default class NumericInput extends FormField {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      caretPosition: 0,
+      separatorsCounter: 0,
```
Comment (DominikGuzei) [inline source/components/NumericInput.js]: Please add some doc comments for the props … for me it's not totally clear what a `separatorsCounter` is 😉

File: stories/NumericInput.stories.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      skin={<SimpleNumericInputSkin />}
+    />
+  ))
+
+  .add('Send amount - maxValue(30000)', () => (
+    <NumericInput
+      label="Amount"
+      rightLabel="ADA"
+      placeholder="0.000000"
+      maxValue={300000000}
```
Comment (DominikGuzei) [inline stories/NumericInput.stories.js]: Btw. the `maxValue` seems to be working incorrectly:

screenshot 2017-05-30 um 11 11 07

Comment (tomothespian) [inline stories/NumericInput.stories.js]: I will see what happens ;)

**Comment by DominikGuzei** [comment]
thanks @tomothespian works perfectly now! 👍

**Pull request merged**